apps/wso2-public-wallet/src/pages/home/home.js

```diff
@@ -0,0 +1,142 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import { message } from "antd";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { getLocalDataAsync } from "../../helpers/storage";
+import { getWalletBalanceByWalletAddress } from "../../services/blockchain.service";
+import {
+  ERROR_RETRIEVE_WALLET_ADDRESS,
+  WALLET_ADDRESS_COPIED,
+  COPIED,
+} from "../../constants/strings";
+import { STORAGE_KEYS } from "../../constants/configs";
+import "./home.css";
+import { useAuthContext } from "@asgardeo/auth-react";
+import NoWallet from "../no-wallet/no-wallet";
+import RecentActivities from "../../components/home/recent-activities";
+import WalletOverview from "../../components/wallet-overview/wallet-overview";
+
+const HomePage = () => {
+  // --- get the navigate function from useNavigate hook ---
+  const navigate = useNavigate();
+
+  // --- get the message api and context holder from the message hook ---
+  const [walletAddress, setWalletAddress] = useState(null);
+  const [isAccountCopied, setIsAccountCopied] = useState(false);
+  const [tokenBalance, setTokenBalance] = useState(0);
+  const [isTokenBalanceLoading, setIsTokenBalanceLoading] = useState(false);
+  const [messageApi, contextHolder] = message.useMessage();
+  const [copied, setCopied] = useState(false);
+
+  // --- get the message api and context holder from the message hook ---
+  const handleCopy = () => {
+    handledCopyAccount();
+    setCopied(true);
+    setTimeout(() => setCopied(false), 2000);
+    message.success(COPIED);
+  };
+
+  // --- fetch wallet address ---
+  const fetchWalletAddress = async () => {
+    try {
+      const walletAddressResponse = await getLocalDataAsync(
+        STORAGE_KEYS.WALLET_ADDRESS
+      );
+      if (walletAddressResponse) {
+        setWalletAddress(walletAddressResponse);
+      }
+    } catch (error) {
+      messageApi.error(ERROR_RETRIEVE_WALLET_ADDRESS);
+    }
+  };
+
+  // --- fetch wallet address when page mounts ---
+  useEffect(() => {
+    fetchWalletAddress();
+  }, []);
+
+  // --- fetch current token balance ---
+  const fetchCurrentTokenBalance = async () => {
+    try {
+      setIsTokenBalanceLoading(true);
+      console.log("this is wallet address --- > ", walletAddress);
+      const tokenBalance = await getWalletBalanceByWalletAddress(walletAddress);
+      console.log("this is token balance --- > ", tokenBalance);
+      setTokenBalance(tokenBalance);
+    } catch (error) {
```

Comment:
```suggestion
    try {
      setIsTokenBalanceLoading(true);
      const tokenBalance = await getWalletBalanceByWalletAddress(walletAddress);
      setTokenBalance(tokenBalance);
    } catch (error) {
```
Check and remove all the `console.log`s except error logs.